tanzu-build-service/tbs-image-signing.md

```diff
@@ -0,0 +1,146 @@
+
+    > Refer to [installing Tanzu Build Service](install-tbs.md) section.
+    The Full, Iterate, and Build profiles include Tanzu Build Service.
+
+2. Cosign
+    > Follow the official docs to [install cosign](https://docs.sigstore.dev/cosign/installation/)
+
+3. A [Builder or ClusterBuilder](https://docs.vmware.com/en/Tanzu-Build-Service/1.5/vmware-tanzu-build-service/GUID-managing-builders.html) and [Image](https://docs.vmware.com/en/Tanzu-Build-Service/1.5/vmware-tanzu-build-service/GUID-managing-images.html) resource configured.
+
+
+### Configure Tanzu Build Service to sign your image builds
+
+1. Generate a cosign key pair
+
+   The `cosign generate-key-pair` command generates a key-pair for you and stores it as a kubernetes secret with name `tutorial-cosign-key-pair` in the `default` namespace.
+
+   ```bash
+   cosign generate-key-pair k8s://default/tutorial-cosign-key-pair
+   ```
+
+   The command will ask you to enter a password for the private key. Enter any password you want. After the command has completed successfully, you should see the following output:
+
+   ```
+   Successfully created secret tutorial-cosign-key-pair in namespace default
+   Public key written to cosign.pub
+   ```
+
+   You should see a `cosign.pub` file in your current folder. Keep this file as it will be needed to verify the signature of the images built in this tutorial.
+
+
+   If you are using [dockerhub](https://hub.docker.com/) or a registry that does not support OCI media types, you need to add the annotation `kpack.io/cosign.docker-media-types: "1"` to the cosign secret. The secret `tutorial-cosign-key-pair` should look something like this:
+
+   ```yaml
+   apiVersion: v1
+   kind: Secret
+   type: Opaque
+   metadata:
+     name: tutorial-cosign-key-pair
+     namespace: default
+     annotations:
+       kpack.io/cosign.docker-media-types: "1"
+   data:
+     cosign.key: <PRIVATE KEY DATA>
+     cosign.password: <COSIGN PASSWORD>
+     cosign.pub: <PUBLIC KEY DATA>
+   ```
+
+    > Note: Learn more about configuring cosign key pairs with the [Tanzu Build Service Image documentation](https://docs.vmware.com/en/Tanzu-Build-Service/1.5/vmware-tanzu-build-service/GUID-managing-images.html#image-signing-with-cosign)
+
+
+2. Create or modify the service account that is referenced in the Image resource so it includes the cosign key pair secret created in the previous step.
+
+   Just by adding a cosign secret to the service account that is referenced in an Image resource enables cosign signing.
+
+   ```yaml
+   apiVersion: v1
+   kind: ServiceAccount
+   metadata:
+     name: tutorial-cosign-service-account
+     namespace: default
+   secrets:
+   - name: registry-credentials
+   - name: tutorial-cosign-key-pair
+   imagePullSecrets:
+   - name: registry-credentials
+   ```
+
+   Apply that service account to the cluster
+
+   ```bash
+   kubectl apply -f cosign-service-account.yaml
+   ```
+
+3. Create an Image resource:
```

Comment:
This should maybe be via a workload ala `tanzu apps workload create` and set the service account via SSC templating?

Reply:
oh I thought this could be done with:

```
tanzu apps workload create <workload-name> \
  --param serviceAccount=<service-account-name>
```

but I understand this may be out of scope!

Reply:
You cannot use `serviceAccount` because that will give the run cluster or deployment namespace access to the private keys used for signing.